### PR TITLE
SUPP: Add black step to CI

### LINKED ISSUE #1966: Add CI steps to check that formatting/linting is correct
We have the pre-commit hook, but we should go further and add a step to the CI pipeline that fails if black would try to make formatting changes to a proposed PR.

### PR DESCRIPTION
**Pull request by toryhaavik**
Closes #1966

### CODE REVIEW

File: ci/requirements-3.6-dev.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 channels:
   - conda-forge
 dependencies:
-  - black
+  - git+git://github.com/psf/black@40e8b3a231bade22d838858435a33d60a8325306#egg=black  # see .pre-commit-config.yaml, we pin to a specific version of black
```
Comment (xmnlab) [inline ci/requirements-3.6-dev.yml]: this probably should be in a pip section.

at the end, inside the pip section after ` - seed-isort-config` you can add:

```
    - git+https://github.com/psf/black.git@40e8b3a231bade22d838858435a33d60a8325306
    - git+https://github.com/timothycrosley/isort.git@18ad293fc9d1852776afe35015a932b68d26fb14
```

also we need a specific version of isort

File: ci/requirements-3.7-dev.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 channels:
   - conda-forge
 dependencies:
-  - black
+  - git+git://github.com/psf/black@40e8b3a231bade22d838858435a33d60a8325306#egg=black  # see .pre-commit-config.yaml, we pin to a specific version of black
```
Comment (xmnlab) [inline ci/requirements-3.7-dev.yml]: see the previous comment about pip section

File: ci/requirements-3.6-dev.yml (hunk 1 above)
Comment (toryhaavik) [reply ci/requirements-3.6-dev.yml]: Thanks! I should have seen the pip section before...

**Comment by toryhaavik** [comment]
Ha! the pipeline failed *correctly* :D. There are some formatting changes that need to be made, i'll add them to this review.

**Comment by xmnlab** [comment]
@toryhaavik that is awesome!

It seems black is not available inside the CI for linux/python 3.5:

```
/activate.sh: line 1: exec: black: not found
```

**Comment by toryhaavik** [comment]
Oh yeah, because black doesn't work on py35. I'll have to add a conditional to check version

**Pull request merged**